app/src/main/java/io/homeassistant/companion/android/widgets/button/ButtonWidgetConfigureActivity.kt

```diff
@@ -75,6 +75,7 @@ class ButtonWidgetConfigureActivity : BaseWidgetConfigureActivity(), IconDialog.
     private lateinit var binding: WidgetButtonConfigureBinding
 
     private var requestLauncherSetup = false
+    private var requireAuthentication = false
```

Comment:
Instead of using a variable here to store state, you can read directly from the database when setting it up (`buttonWidget.requireAuthentication`), and read directly from the UI when saving it (`binding.widgetCheckboxRequireAuthentication.isChecked`). This better matches how other other saved values like title and icon are handled.